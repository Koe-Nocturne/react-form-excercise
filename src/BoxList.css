.box {
    margin: 10px;
    border-radius: 5px;
}

.boxContainer {
    display: flex;
    justify-content: center;
}
