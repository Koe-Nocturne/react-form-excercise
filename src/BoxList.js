import React from 'react';
import NewBoxForm from './NewBoxForm'
import uuid from 'uuid/v4';
import Box from './Box';


class BoxList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      boxes: []
    }
    this.onSubmit = this.onSubmit.bind(this);
    this.deleteBox = this.deleteBox.bind(this);
  }

  createBox() {
    return (
      <div className="boxContainer">
        {this.state.boxes.map(box =>
          <Box key={box.id} backgroundColor={box.backgroundColor}
            width={box.width}
            height={box.height} removeBox={this.deleteBox} />)}
      </div>

    )


  }

  handleChange(evt) {
    this.setState({ [evt.target]: evt.target.value })
  }

  deleteBox(id) {
    debugger;
    this.setState(state => ({
      boxes: state.boxes.filter(key => key.id !== id)
    })
    )
  }

  onSubmit(boxData) {
    let newBox = { ...boxData, id: uuid() };
    this.setState(state => ({
      boxes: [...state.boxes, newBox]
    }));

  }

  render() {
    return (
      <div>
        <NewBoxForm onSubmit={this.onSubmit} />
        {this.createBox()}
      </div>
    );
  }
}

export default BoxList;
