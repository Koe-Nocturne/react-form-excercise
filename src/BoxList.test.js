import React from 'react';
import BoxList from './BoxList';
import { shallow, mount } from 'enzyme';

describe("boxList renders", () =>{
  it("renders correctly", () => {
    let wrapper = shallow(<BoxList/>);

    expect(wrapper).toMatchSnapshot();
  });
  it("renders a box correctly", () => {
    let wrapper = mount(<BoxList/>);

    wrapper.setState({boxes: [{backgroundColor: "blueviolet", height: "150", width:"250", id: "abc"}]})

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().boxes.length).toEqual(1);
    expect(wrapper.state().boxes[0]).toEqual({backgroundColor: "blueviolet", height: "150", width:"250", id: expect.any(String)});
  });
});

describe("delete box", () => {
  it("can we delete a box", () => {
    let wrapper = mount(<BoxList/>);

    wrapper.setState({boxes: [{backgroundColor: "blueviolet", height: "150", width:"250", id: "abc"}]})

    wrapper.find('button').at(1).simulate('click');
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().boxes.length).toEqual(0);
    expect(wrapper.state().boxes[0]).toEqual(undefined);
  });
});
