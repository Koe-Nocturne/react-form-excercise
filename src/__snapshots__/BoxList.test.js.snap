// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boxList renders renders a box correctly 1`] = `ReactWrapper {}`;

exports[`boxList renders renders correctly 1`] = `ShallowWrapper {}`;

exports[`delete box can we delete a box 1`] = `ReactWrapper {}`;
